Makefile

```diff
@@ -1,33 +1,52 @@
-.PHONY: build run default
+.PHONY: build run default test doc dg upstream downstream
 
-TLS=TLS
-IMAP=IMAP
-IMAGE_NAME = postfix
-IMAGE_NAME_TLS = postfix-`echo $(TLS) | tr '[:upper:]' '[:lower:]'`
-IMAGE_NAME_IMAP = postfix-`echo $(IMAP) | tr '[:upper:]' '[:lower:]'`
-DOCKER_TLS=Dockerfile.$(TLS)
-DOCKER_IMAP=Dockerfile.$(IMAP)
+VARIANT := upstream
+DISTRO = fedora-26-x86_64
+DG = /usr/bin/dg
 
-all: run run_tls run_imap
-default: run
+DG_EXEC = ${DG} --max-passes 25 --distro ${DISTRO}.yaml --spec specs/configuration.yml --multispec specs/multispec.yml --multispec-selector variant=$(VARIANT)
+DISTRO_ID = $(shell ${DG_EXEC} --template "{{ config.os.id }}")
 
-build:
-	docker build --tag=$(IMAGE_NAME) .
+IMAGE_REPOSITORY = $(shell ${DG_EXEC} --template "{{ spec.image_repository }}")
 
-build_tls:
-	docker build -f $(DOCKER_TLS) --tag=$(IMAGE_NAME_TLS) .
+default: run
+
+all: run run_tls run_imap
 
-build_imap:
-	docker build -f $(DOCKER_IMAP) --tag=$(IMAGE_NAME_IMAP) .
+build: doc
+	docker build --tag=$(IMAGE_REPOSITORY) -f Dockerfile.rendered .
 
 run: build
-	docker run -p 25:10025 -e MYHOSTNAME=localhost $(IMAGE_NAME)
+	docker run -p 25:10025 -e MYHOSTNAME=localhost $(IMAGE_REPOSITORY)
+
+run_tls:
+	docker run -p 25:10025 -e ENABLE_TLS -e SMTP_USER=user:pwd -e MYHOSTNAME=localhost $(IMAGE_REPOSITORY)
+
+run_imap:
+	docker run -p 587:10587 -e ENABLE_IMAP -e MYHOSTNAME=localhost -e SMTP_USER=user:pwd -v <path_to_certificates>:/etc/postfix/certs $(IMAGE_REPOSITORY)
```

Comment:
How is path_to_certificates determined here? Should we pass it in or use a default path?

Reply:
It depends on user. I would prefer the default path.

Reply:
What's the point to have this command in makefile when it *never* succeeds? `path_to_certificates` should be a make variable.